src/content/developers/docs/oracles/index.md

```diff
@@ -1,314 +1,86 @@
-     */
-    constructor() public {
-        priceFeed = AggregatorV3Interface(0x9326BFA02ADD2366b30bacB125260Af641031331);
-    }
-
-    /**
-     * Returns the latest price
-     */
-    function getLatestPrice() public view returns (int) {
-        (
-            uint80 roundID,
-            int price,
-            uint startedAt,
-            uint timeStamp,
-            uint80 answeredInRound
-        ) = priceFeed.latestRoundData();
-        return price;
-    }
-}
-```
+## Prerequisites {#prerequisites}
 
-[You can test this in remix with this link](https://remix.ethereum.org/#version=soljson-v0.6.7+commit.b8d736ae.js&optimize=false&evmVersion=null&gist=0c5928a00094810d2ba01fd8d1083581)
+This page assumes the reader is familiar with Ethereum fundamentals, including [nodes](/developers/docs/nodes-and-clients/), [consensus mechanisms](/developers/docs/consensus-mechanisms/), and the [EVM](/developers/docs/evm/). You should also have a good grasp of [smart contracts](/developers/docs/smart-contracts/) and [smart contract anatomy](/developers/docs/smart-contracts/anatomy/), especially events. 
 
-[View the docs](https://docs.chain.link/docs/get-the-latest-price)
+## What is a blockchain oracle? {#what-is-a-blockchain-oracle} 
 
-### Chainlink VRF {#chainlink-vrf}
+Oracles are applications that source, verify, and transmit external information (i.e., information stored off-chain) to smart contracts running on the blockchain. Besides “pulling” off-chain data and broadcasting it on Ethereum, oracles can also “push” information from the blockchain to external systems. An example is an oracle that unlocks a smart lock once the user sends the fee via an Ethereum transaction. 
 
-Chainlink VRF (Verifiable Random Function) is a provably-fair and verifiable source of randomness designed for smart contracts. Smart contract developers can use Chainlink VRF as a tamper-proof random number generation (RNG) to build reliable smart contracts for any applications which rely on unpredictable outcomes:
+Oracles act as a “bridge” connecting smart contracts and blockchains to the real world. Without oracles, smart contract applications would be limited in usefulness and real-world value. A blockchain oracle provides a mechanism for extracting real-world information which smart contracts can use to trigger state changes in the [Ethereum Virtual Machine](/developers/docs/evm/)
 
-- Blockchain games and NFTs
-- Random assignment of duties and resources (e.g. randomly assigning judges to cases)
-- Choosing a representative sample for consensus mechanisms
+Oracles differ based on the source of data (one or multiple sources), direction of information (inbound or outbound), trust models (centralized or decentralized) and system architecture (immediate-read, publish-subscribe, and request-response). We can also distinguish between oracles that only retrieve external data (input oracles) and ones that perform computation off-chain (computational oracles).
```

Comment:
I don;t understand the "direction of information" - surely an oracle must have inbound data (real world -> oracle) and outbound data (oracle -> contract) to be useful?

Reply:
Wanted to distinguish between an oracle that "pushes" information from the blockchain to the real world, e.g., one that triggers a smart lock if the user sends an on-chain transaction and an oracle that "pulls" information to the blockchain (e.g., price oracles). That said, I think "input vs output oracles" is a better way to highlight this detail--will change now,